[Workers] env variables revision (#2103)
cloudflare/cloudflare-docs: products/workers/src/content/platform/environment-variables.md

---
order: 
pcx-content-type: concept
---

# Environment variables

In the Workers platform, environment variables, secrets, and KV namespaces are known as bindings. Regardless of type, bindings are always available as global variables within your Worker script.

## Adding environment variables via wrangler

> * Environment variables are defined via the `[vars]` configuration in your `wranger.toml` file and are always plaintext values.

Reviewer: For SEO and scanability, I'd change this from a bulleted list to some `h3`s

    ```toml
    ---
    filename: wrangler.toml
    ---
    name = "my-worker-dev"
    type = "javascript"

    account_id = "<YOUR ACCOUNTID>"
    workers_dev = true

    # [secrets]
    # SPARKPOST_KEY
    # GTOKEN_PRIVKEY
    # GTOKEN_KID
    ```

<Aside type="warning">

\* __Warning:__ Do not use plaintext environment variables to store sensitive information. Use [`wrangler secret put`](/cli-wrangler/commands#secret) instead.

</Aside>

## Adding environment variables via the dashboard

Add environment variables by logging into [Cloudflare dashboard](https://dash.cloudflare.com/) > **Account Home** > **Workers** and select your **Workers script**. 

To add environment variables, such as `vars` and `secret`:

1. Go to your **Workers script** > **Settings** > **Add variable** under **Environment Variables**.
2. Input a **Variable name** and its **value**, which will be made available to your Worker. 

> 3. If your variable is a secret select  **Encrypt** to protect its value. This will prevent the value from being visible via `wrangler` and the dashboard. 

Reviewer: ```suggestion
3. If your variable is a secret, click **Encrypt** to protect its value. This will prevent the value from being visible via `wrangler` and the dashboard. 
```

> 3. (Optional) To add multiple environment variables, select **Add variable**. 

Reviewer: ```suggestion
3. (Optional) To add multiple environment variables, click **Add variable**. 
```

> 5. Select **Save** to implement your changes.

Reviewer: ```suggestion
5. Click **Save** to implement your changes.
```

Reviewer: Change in other places too. I think we decided on `Click` for buttons and Select/Choose for items in a dropdown list.

![env variables dash](env_variables_dash.png)

To add KV namespace bindings:

1. Go to your **Workers script** > **Settings** > **Add binding** under **KV Namespace Bindings**.
2. Choose a **Variable name**. This will be the way the variable name will be referenced in your Worker script. 

> 3. Next, select a **KV namespace** from the dropdown. 

Author: ```suggestion
3. Next, select a **KV name space** from the dropdown. 
```

4. Select **Add binding** to add multiple bindings. 
5. When you are finished, select **Save** to implement your changes.

![kv namespace bindings](kv_namespace_bindings.png)

> Your completed Workers dashboard, with environment variables and KV namespace bindings added, will look like the following example reference. 

Author: ```suggestion
## Your completed Workers dashboard, with environment variables and KV namespace bindings added, will look like the following example reference. 
```
